index.js

```diff
@@ -0,0 +1,299 @@
+(function() {
+  function Calculator(domElement) {
+    this.current = domElement,
+    this.display = this.current.querySelector('.calculator-valueDisplay')
+    this.first = 0,
+    this.second = 0,
+    this.operand = '',
+    this.result = '',
+    this.firstValueAdded = false,
+    this.secondValueAdded = false
+  }
+
+  Calculator.prototype.addTheListeners = function() {
+    var calculator = this
+    var currentCalc = this.current
+    currentCalc.addEventListener('click', function(event){
+      if(event.target.matches('.calculator-button:not(#flip):not(#percentage), #decimal')){
+        calculator.displayValue(event.target.innerHTML)
+      }else if(event.target.matches('#divide, #multiply, #subtract, #add')){
+        calculator.getOperand(event.target.innerHTML)
+      }else if(event.target.matches('#AC')){
+        calculator.allClear()
+      }else if(event.target.matches('#percentage')){
+        calculator.percent()
+      }else if(event.target.matches('#equals')){
+        calculator.calculate()
+      }else if(event.target.matches('#flip')){
+        calculator.flipSign()
+      }
+    })
+    currentCalc.onkeydown = function(event){
+      if(parseInt(event.key) || event.key == 0){
+        calculator.assignClass(event.key)
+        calculator.displayValue(event.key)
+      }
+      if(calculator.isOperand(event.key)) {
+        calculator.assignClass(event.key)
+        calculator.getOperand(event.key)
+      }
+      if(event.key == '=') {
+        calculator.assignClass(event.key)
+        calculator.calculate()
+      }
+      if(event.key == '.') {
+        calculator.assignClass(event.key)
+        calculator.displayValue('.')
+      }
+      if(event.key == 'Clear') {
+        calculator.assignClass(event.key)
+        calculator.allClear()
+      }
+    }
+  }
+
+  Calculator.prototype.changeFontSize = function(){
+    var calculator = this
+    var maxLength = calculator.display
+    var currentLength = calculator.display.innerHTML
+    if(currentLength.length >= maxLength.attributes.maxlength.value - 2){
+      maxLength.style.fontSize = "30px"
+    }else{
+      maxLength.style.fontSize = "70px"
+    }
+  }
+  Calculator.prototype.displayValue = function(num){
+    var calculator = this
+    if(calculator.secondValueAdded){
+      calculator.firstValueAdded = false
+      calculator.secondValueAdded = false
+      calculator.first = calculator.result
+      calculator.second = 0
+      calculator.display.innerHTML = 0
+    }
+    calculator.changeFontSize()
+    if(!calculator.firstValueAdded){
+      if(calculator.display.innerHTML == 0){
+        calculator.display.innerHTML = null
```

Comment:
`calculator.display.innerHTML = null` is weird. Try `calculator.display.innerText = ''`

you should be using `.innerText = ` instead of `.innerHTML` since your providing plain text values